### PR TITLE
Issue 345: Add priors to `epinowcast` object

### LINKED ISSUE #345: Add priors to epinowcast object
**Is your feature request related to a problem? Please describe.**
At the moment the `epinowcast` function doesn't store the precise priors that it uses when fitting a model. For inspection, reproducibility it would be useful if it did.

**Describe the solution you'd like**
I would like a new variable to be added to the `epinowcast` object that contains a `data.table` of the priors used for model fitting.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Description

This PR closes #345 by adding the defined or default priors to the returned `epinowcast` object.

[Describe the changes that you made in this pull request.]

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: R/epinowcast.R [hunk 1]
```diff
@@ -190,7 +190,7 @@ epinowcast <- function(data,
     )
   )
 
-  out <- cbind(data, fit)
+  out <- cbind(data, fit, priors = list(priors))
```
Comment (seabbs) [inline R/epinowcast.R:193]: we might want priors to appear before the fit output for the flow to work?

**Review by seabbs — COMMENTED** [review]
Thanks for this @jamesmbaazam - this is looking good so far. See comment about ordering of epinowcast output.

File: R/epinowcast.R (hunk 1 above)
Comment (jamesmbaazam) [reply R/epinowcast.R]: See change in e277b5b83ce09cf695e797cc413ffce324a9b7b1.

**Review by seabbs — COMMENTED** [review]
Thanks @jamesmbaazam, this looks good to me. Just needs a news item and we can merge. As this is your second PR and you've been active on reviews etc. I think you have also crossed the author threshold (@pearsonca / @sbfnk / @adrianlison confirm?)? Happy to change your status with this PR as we;;?

**Comment by jamesmbaazam** [comment]
> Happy to change your status with this PR as we;;?

I'm planning to work on #323 early this week. We can add me as an author then if my authorship status is currently arguable ☺️ .

File: NEWS.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ This release is in development and not yet ready for production use.
...
 - Added support for using a maximum delay that is longer than the largest observed delay in the data. This may be useful at the start of an outbreak, when the data is sparse and the user expects delays longer than what has been observed so far. Note that because this requires extrapolating the delay distribution beyond the support of the data, users should be cautious when using this feature. A new example, `inst/examples/germany_max_delay_greater_than_data.R`, has been added to demonstrate this feature. See # by @seabbs and self-reviewed.
+- Added the priors used for model fitting to the `<epinowcast>` object. The object returned by `epinowcast()` now has a variable called `priors` and can be accessed for inspection and downstream analyses. See #399 and reviewed by @seabbs.
```
Comment (seabbs) [inline NEWS.md:47]:
```suggestion
- Added the priors used for model fitting to the `<epinowcast>` object. The object returned by `epinowcast()` now has a variable called `priors` and can be accessed for inspection and downstream analyses. See #399 by @jamesmbaazam and reviewed by @seabbs.
```

**Review by seabbs — APPROVED** [review]
LGTM. Slightly tweaked the news item to add clearer authorship.

File: NEWS.md (hunk 2 above)
Comment (jamesmbaazam) [reply NEWS.md]: Oops. Missed that. Thanks.

File: tests/testthat/test-epinowcast.R
Comment (pearsonca) [inline tests/testthat/test-epinowcast.R:1]: Seems like the tests should have expanded not just to now-also-ignore priors in comparisons, but to check whether or not some were properly included.

**Review by pearsonca — COMMENTED** [review]
Doesn't seem like the unit testing bit is actually testing if the priors are what's offered via arguments.

File: tests/testthat/test-epinowcast.R
Comment (pearsonca) [reply tests/testthat/test-epinowcast.R:1]: To clarify, I think there should be a test of the character of lines 47-51 in `test-epinowcast.R` added to that general region. That kind of check is also repeated in at least one of the other tests, so need to look around a bit.

Comment (jamesmbaazam) [reply tests/testthat/test-epinowcast.R:1]: Thanks, Carl. It was mostly an omission. I didn't add new tests. Only fixed what was broken. I will add specific tests around the prior object.

**Comment by jamesmbaazam** [comment]
@seabbs I've updated the tests to check aspects of the structure of the expected prior used. It's made me realise that the documentation of the `prior` argument in `epinowcast()` could be fleshed out further because it's not clear to me, from a user perspective, how to specify my own prior, especially the "variable" column in the expected `data.frame`.

I also had to remember to recompile the Getting Started vignette to update the section on the "epinowcast object". Do we maybe need a bullet point in the PR template to remind contributors to recompile vignettes that will potentially be impacted by changes being introduced?

File: NEWS.md [hunk 3]
```diff
@@ -44,6 +44,7 @@ This release is in development and not yet ready for production use.
...
 - Added support for using a maximum delay that is longer than the largest observed delay in the data. This may be useful at the start of an outbreak, when the data is sparse and the user expects delays longer than what has been observed so far. Note that because this requires extrapolating the delay distribution beyond the support of the data, users should be cautious when using this feature. A new example, `inst/examples/germany_max_delay_greater_than_data.R`, has been added to demonstrate this feature. See # by @seabbs and self-reviewed.
+- Added the priors used for model fitting to the `<epinowcast>` object. The object returned by `epinowcast()` now has a variable called `priors` and can be accessed for inspection and downstream analyses. See #399 by @jamesmbaazam and reviewed by @seabbs.
```
Comment (seabbs) [inline NEWS.md:47]:
```suggestion
- Added the priors used for model fitting to the `<epinowcast>` object. The object returned by `epinowcast()` now has a variable called `priors` and can be accessed for inspection and downstream analyses. See #399 by @jamesmbaazam and reviewed by @pearsonca  and @seabbs.
```

**Review by seabbs — APPROVED** [review]
This looks good to me I think - just need to add @pearsonca to the review note.

File: vignettes/epinowcast.Rmd [hunk 4]
```diff
@@ -26,7 +26,7 @@ In this quick start, we demonstrate using `epinowcast` to specify and fit a mini
 
 # Package
 
-In this quick start, we also use `data.table` and `ggplot2` packages. These are both installed as dependencies when `epinowcast` is installed. Note that all output from `epinowcast` is readily useable with  other tools, including `tidyverse` packages (see [here](https://mgimond.github.io/rug_2019_12/Index.html) for a comparison).
```
Comment (seabbs) [inline vignettes/epinowcast.Rmd:29]: Ah damn. Can you revert this and I will fix by adding these changes to the non-rendered version? It looks like the recent PR (#406) accidently targeted the wrong thing.

**Comment by seabbs** [comment]
> @seabbs I've updated the tests to check aspects of the structure of the expected prior used. It's made me realise that the documentation of the prior argument in epinowcast() could be fleshed out further because it's not clear to me, from a user perspective, how to specify my own prior, especially the "variable" column in the expected data.frame.

Okay interesting. To me this is clear but clearly we should update. Can we make a new issue to workshop something?

**Comment by seabbs** [comment]
> I also had to remember to recompile the Getting Started vignette to update the section on the "epinowcast object". Do we maybe need a bullet point in the PR template to remind contributors to recompile vignettes that will potentially be impacted by changes being introduced?

Indeed this is actually because #406 accidently targetted the wrong thing and the reviewer (me) didn't notice. So we need to make it clear to target the precompiled vignettes if present and to compile them (though this may be not needed if we start doing that in the CI #379)

Also note that if we can keep changes to the templates etc as relatively generic then we can share them across the organisation (i.e #371 - which is in progress by @pearsonca)

File: vignettes/epinowcast.Rmd (hunk 4 above)
Comment (jamesmbaazam) [reply vignettes/epinowcast.Rmd]: Alright. See fc18c2680d3645822c8aa5adefa5748864ae3ca1.

**Review by seabbs — APPROVED** [review]
LGTM

**Review by seabbs — APPROVED** [review]
LGTM

**Pull request merged**